test/parallel/test-net-connect-immediate-destroy.js

```diff
@@ -2,7 +2,11 @@
 const common = require('../common');
 const net = require('net');
 
-const socket = net.connect(common.PORT, common.localhostIPv4,
-                           common.mustNotCall());
-socket.on('error', common.mustNotCall());
-socket.destroy();
+const s = net.createServer();
+s.listen(0);
+const port = s.address().port;
+s.close(() => {
+  const socket = net.connect(port, common.localhostIPv4, common.mustNotCall());
+  socket.on('error', common.mustNotCall());
+  socket.destroy();
+});
```

Comment:
I don't think it's a silly question at all. I _think_ that closing the socket right after calling connect should work because the connect operation is asynchronous and we're closing the socket synchronously (I haven't tested on every platform though just locally on `OS X`, I started a CI job with that changes to see if it works: https://ci.nodejs.org/job/node-test-commit/9693/https://ci.nodejs.org/job/node-test-commit/9693/). OTOH I would argue that creating a server just to reserve a port might make the code too convoluted and moving it to sequential could be a better option.